modin/pandas/groupby.py

```diff
@@ -325,37 +447,31 @@ def __getitem__(self, key):
         NotImplementedError
             Column lookups on GroupBy with arbitrary Series in by is not yet supported.
         """
+        if self._selection is not None:
+            raise IndexError(f"Column(s) {self._selection} already selected")
+
         kwargs = {**self._kwargs.copy(), "squeeze": self._squeeze}
         # Most of time indexing DataFrameGroupBy results in another DataFrameGroupBy object unless circumstances are
         # special in which case SeriesGroupBy has to be returned. Such circumstances are when key equals to a single
         # column name and is not a list of column names or list of one column name.
-        make_dataframe = True
-        if self._drop and self._as_index:
-            if not isinstance(key, list):
-                key = [key]
+        if is_list_like(key):
+            make_dataframe = True
+        else:
+            if self._as_index:
                 kwargs["squeeze"] = True
                 make_dataframe = False
-        # When `as_index` is False, pandas will always convert to a `DataFrame`, we
-        # convert to a list here so that the result will be a `DataFrame`.
-        elif not self._as_index and not isinstance(key, list):
-            # Sometimes `__getitem__` doesn't only get the item, it also gets the `by`
-            # column. This logic is here to ensure that we also get the `by` data so
-            # that it is there for `as_index=False`.
-            if (
-                isinstance(self._by, type(self._query_compiler))
-                and all(c in self._columns for c in self._by.columns)
-                and self._drop
-            ):
-                key = list(self._by.columns) + [key]
             else:
+                kwargs["ndim"] = 1
```

Comment:
Why is `ndim` equal to 1?

Reply:
answered [here](https://github.com/modin-project/modin/pull/3298#discussion_r694241989)